### PR TITLE
Make Jwk.alg optional, add did:web localhost resolution support

### PR DESCRIPTION
**Pull request by KendallWeihe**

### CODE REVIEW

**Comment by KendallWeihe** [comment]
This PR doesn't include binding changes, because we need to first implement build systems for multi-targets as we did in `tbdex-rs` here https://github.com/TBD54566975/tbdex-rs/pull/67

File: crates/web5/src/dids/methods/did_web/resolver.rs [hunk 1]
```diff
@@ -34,7 +34,14 @@ impl Resolver {
         };
 
         Self {
-            did_url: format!("https://{}/did.json", did_url),
+            did_url: format!(
+                "{}://{}/did.json",
+                match did_url.contains("localhost") {
```
Comment (nitro-neal) [inline crates/web5/src/dids/methods/did_web/resolver.rs:39]: should this be == localhost?
What if I have neal-localhost.com ?

**Review by nitro-neal — APPROVED** [review]
we need to `just bind` and fix kt after this right?

File: crates/web5/src/dids/methods/did_web/resolver.rs (hunk 1 above)
Comment (KendallWeihe) [reply crates/web5/src/dids/methods/did_web/resolver.rs:39]: yeah we need to use the `Url` crate, but will do subsequently

**Pull request merged**